In [59]:
import sys
import pandas as pd
import os


In [60]:
sys.path.append('../src')  # utils.py'yi çağırmak için

In [62]:
print(os.listdir('../src'))

['__pycache__']


In [63]:
from utils import load_dataset, check_missing_values


dataset_inceleme.pynb dosyasında okudugum verisetlerini buraya cektim 

In [64]:
df_orders = load_dataset('../data/olist_orders_dataset.csv')
df_customers = load_dataset('../data/olist_customers_dataset.csv')
df_items = load_dataset('../data/olist_order_items_dataset.csv')
df_reviews = load_dataset('../data/olist_order_reviews_dataset.csv')
df_products = load_dataset('../data/olist_products_dataset.csv')
df_sellers = load_dataset('../data/olist_sellers_dataset.csv')
df_geolocation = load_dataset('../data/olist_geolocation_dataset.csv')
df_category_translation = load_dataset('../data/product_category_name_translation.csv')
df_payments = load_dataset('../data/olist_order_payments_dataset.csv')  # Eksik olan dataset


'../data/olist_orders_dataset.csv' başarıyla yüklendi. Şekil: (99441, 8)
'../data/olist_customers_dataset.csv' başarıyla yüklendi. Şekil: (99441, 5)
'../data/olist_order_items_dataset.csv' başarıyla yüklendi. Şekil: (112650, 7)
'../data/olist_order_reviews_dataset.csv' başarıyla yüklendi. Şekil: (99224, 7)
'../data/olist_products_dataset.csv' başarıyla yüklendi. Şekil: (32951, 9)
'../data/olist_sellers_dataset.csv' başarıyla yüklendi. Şekil: (3095, 4)
'../data/olist_geolocation_dataset.csv' başarıyla yüklendi. Şekil: (1000163, 5)
'../data/product_category_name_translation.csv' başarıyla yüklendi. Şekil: (71, 2)
'../data/olist_order_payments_dataset.csv' başarıyla yüklendi. Şekil: (103886, 5)


In [65]:
df_orders.to_pickle("../data/df_orders.pkl")
df_customers.to_pickle("../data/df_customers.pkl")
df_items.to_pickle("../data/df_items.pkl")
df_reviews.to_pickle("../data/df_reviews.pkl")
df_products.to_pickle("../data/df_products.pkl")
df_sellers.to_pickle("../data/df_sellers.pkl")
df_geolocation.to_pickle("../data/df_geolocation.pkl")
df_category_translation.to_pickle("../data/df_category_translation.pkl")
df_payments.to_pickle("../data/df_payments.pkl")


----------

df_orders veriseti

In [66]:
df_orders.head()


order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
1  53cdb2fc8bc7dce0b6741e2150273451  b0830fb4747a6c6d20dea0b8c802d7ef   
2  47770eb9100c2d0c44946d9cf07ec65d  41ce2a54c0b03bf3443c3d931a367089   
3  949d5b44dbf5de918fe9c16f97b45f8a  f88197465ea7920adcdbec7375364d82   
4  ad21c59c0840e6cb83a9ceb5573f8159  8ab97904e6daea8866dbdbc4fb7aad2c   

  order_status order_purchase_timestamp    order_approved_at  \
0    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
1    delivered      2018-07-24 20:41:37  2018-07-26 03:24:27   
2    delivered      2018-08-08 08:38:49  2018-08-08 08:55:23   
3    delivered      2017-11-18 19:28:06  2017-11-18 19:45:59   
4    delivered      2018-02-13 21:18:39  2018-02-13 22:20:29   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2017-10-10 21:25:13   
1          2018-07-26 14:31:00           2018-08-07 15:27:45   
2          2018-08-08 13:50:00           2018-08-17 18:06:29   
3          2017-11-22 13:39:59           2017-12-02 00:28:42   
4          2018-02-14 19:46:34           2018-02-16 18:17:02   

  order_estimated_delivery_date  
0           2017-10-18 00:00:00  
1           2018-08-13 00:00:00  
2           2018-09-04 00:00:00  
3           2017-12-15 00:00:00  
4           2018-02-26 00:00:00

In [67]:
df_orders.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 8 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   order_id                       99441 non-null  object
 1   customer_id                    99441 non-null  object
 2   order_status                   99441 non-null  object
 3   order_purchase_timestamp       99441 non-null  object
 4   order_approved_at              99281 non-null  object
 5   order_delivered_carrier_date   97658 non-null  object
 6   order_delivered_customer_date  96476 non-null  object
 7   order_estimated_delivery_date  99441 non-null  object
dtypes: object(8)
memory usage: 6.1+ MB


In [27]:
check_missing_values(df_orders)

Eksik Değer Sayısı  Yüzde (%)
order_delivered_customer_date                2965       2.98
order_delivered_carrier_date                 1783       1.79
order_approved_at                             160       0.16

head:Sipariş ID’si, müşteri ID, sipariş tarihi, sipariş durumu gibi bilgiler var.

info: Tarihsel kolonlar: purchase, delivered, estimated vs. arih bilgisi içeren sütunları pd.to_datetime() ile datetime formatına çevirmek analiz ve görselleştirme için faydalı olur.

check_missing_values:Eksik değerlerin nedenlerini araştırabiliriz (iptal edilen siparişler mi, henüz işlenmemiş siparişler mi vs.).
Bu sütunları datetime tipine dönüştürüp analizlere başlayabiliriz.
İstersen bu fonksiyon çıktısına göre eksik verilerin nasıl işleneceğine karar verebiliriz.

Buradan siparişlerin zaman içindeki dağılımını, durumu (teslim, iptal, vb) analiz edebiliriz.

Neler Yapabiliriz?
Siparişlerin genel durumu:

Hangi sipariş durumu (delivered, canceled, shipped vs) ne kadar yaygın?

Sipariş durumu bazında zaman analizi (örneğin onaylanma süresi, teslim süresi)

Zaman Analizi:

Siparişlerin satın alındığı tarihlerin dağılımı (mevsimsel, aylık, haftalık)

Siparişlerin onaylanma, teslim edilme süreleri (order_purchase_timestamp ile order_delivered_customer_date arasındaki fark)

Müşteri bazında sipariş sayısı:

Müşterilerin toplam sipariş sayıları

Aktif müşteriler (en çok sipariş verenler)

Teslimat gecikmeleri:

Tahmini teslimat ve gerçek teslimat tarihi farkları

Ortalama teslimat süresi

Sipariş iptalleri:

İptal edilen sipariş oranları ve iptal zamanları


In [29]:
df_customers.head()


customer_id                customer_unique_id  \
0  06b8999e2fba1a1fbc88172c00ba8bc7  861eff4711a542e4b93843c6dd7febb0   
1  18955e83d337fd6b2def6b18a428ac77  290c77bc529b7ac935b93aa66c333dc3   
2  4e7b3e00288586ebd08712fdd0374a03  060e732b5b29e8181a18229c7b0b2b5e   
3  b2b6027bc5c5109e529d4dc6358b12c3  259dac757896d24d7702b9acbbff3f3c   
4  4f2d8ab171c80ec8364f7c12e35b23ad  345ecd01c38d18a9036ed96c73b8d066   

   customer_zip_code_prefix          customer_city customer_state  
0                     14409                 franca             SP  
1                      9790  sao bernardo do campo             SP  
2                      1151              sao paulo             SP  
3                      8775        mogi das cruzes             SP  
4                     13056               campinas             SP

In [30]:
df_customers.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   customer_id               99441 non-null  object
 1   customer_unique_id        99441 non-null  object
 2   customer_zip_code_prefix  99441 non-null  int64 
 3   customer_city             99441 non-null  object
 4   customer_state            99441 non-null  object
dtypes: int64(1), object(4)
memory usage: 3.8+ MB


In [31]:
check_missing_values(df_customers)

Empty DataFrame
Columns: [Eksik Değer Sayısı, Yüzde (%)]
Index: []

Müşteri bilgileri eksiksiz.

Müşteri şehirleri ve eyaletleri var, bunlar demografik analiz için kullanışlı.

Aynı müşteri farklı customer_id ile birden fazla kayıt olabilir (bunu ileride sorgulayabiliriz).

Hiç eksik veri yok.
Neler Yapabiliriz?
Coğrafi Analiz:

Müşterilerin dağılımı şehir ve eyalet bazında

Yoğun müşteri bölgeleri

Müşteri Tekrarı:

unique_id ile aynı müşterinin kaç sipariş verdiği (müşteri tekrar analizi)

Müşteri Segmentasyonu:

Bölgelere göre müşteri segmentleri oluşturmak

Şehir veya eyalet bazlı alışveriş alışkanlıkları

Zip kodu analizleri:

Zip kodlarına göre müşteri sayısı ve sipariş hacmi

In [35]:
df_items.head(10)


order_id  order_item_id  \
0  00010242fe8c5a6d1ba2dd792cb16214              1   
1  00018f77f2f0320c557190d7a144bdd3              1   
2  000229ec398224ef6ca0657da4fc703e              1   
3  00024acbcdf0a6daa1e931b038114c75              1   
4  00042b26cf59d7ce69dfabb4e55b4fd9              1   
5  00048cc3ae777c65dbb7d2a0634bc1ea              1   
6  00054e8431b9d7675808bcb819fb4a32              1   
7  000576fe39319847cbb9d288c5617fa6              1   
8  0005a1a1728c9d785b8e2b08b904576c              1   
9  0005f50442cb953dcd1d21e1fb923495              1   

                         product_id                         seller_id  \
0  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
1  e5f2d52b802189ee658865ca93d83a8f  dd7ddc04e1b6c2c614352b383efe2d36   
2  c777355d18b72b67abbeef9df44fd0fd  5b51032eddd242adc84c38acab88f23d   
3  7634da152a4610f1595efa32f14722fc  9d7a1d34a5052409006425275ba1c2b4   
4  ac6c3623068f30de03045865e4e10089  df560393f3a51e74553ab94004ba5c87   
5  ef92defde845ab8450f9d70c526ef70f  6426d21aca402a131fc0a5d0960a3c90   
6  8d4f2bb7e93e6710a28f34fa83ee7d28  7040e82f899a04d1b434b795a43b4617   
7  557d850972a7d6f792fd18ae1400d9b6  5996cddab893a4652a15592fb58ab8db   
8  310ae3c140ff94b03219ad0adc3c778f  a416b6a846a11724393025641d4edd5e   
9  4535b0e1091c278dfd193e5a1d63b39f  ba143b05f0110f0dc71ad71b4466ce92   

   shipping_limit_date   price  freight_value  
0  2017-09-19 09:45:35   58.90          13.29  
1  2017-05-03 11:05:13  239.90          19.93  
2  2018-01-18 14:48:30  199.00          17.87  
3  2018-08-15 10:10:18   12.99          12.79  
4  2017-02-13 13:57:51  199.90          18.14  
5  2017-05-23 03:55:27   21.90          12.69  
6  2017-12-14 12:10:31   19.90          11.85  
7  2018-07-10 12:30:45  810.00          70.75  
8  2018-03-26 18:31:29  145.95          11.65  
9  2018-07-06 14:10:56   53.99          11.40

In [33]:
df_items.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112650 entries, 0 to 112649
Data columns (total 7 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   order_id             112650 non-null  object 
 1   order_item_id        112650 non-null  int64  
 2   product_id           112650 non-null  object 
 3   seller_id            112650 non-null  object 
 4   shipping_limit_date  112650 non-null  object 
 5   price                112650 non-null  float64
 6   freight_value        112650 non-null  float64
dtypes: float64(2), int64(1), object(4)
memory usage: 6.0+ MB


In [34]:
check_missing_values(df_items)

Empty DataFrame
Columns: [Eksik Değer Sayısı, Yüzde (%)]
Index: []

Eksik değer yok.

112.650 ürün satırı var.

shipping_limit_date datetime 
Fiyat ve kargo ücreti analizi: Ortalama, medyan, dağılım.

Ürün bazında satış analizi: En çok satılan ürünler (ürün_id bazında satış adetleri).

Satıcı bazında performans: Her satıcı kaç ürün satmış, toplam ciro.

Sipariş bazında ürün sayısı: Bir siparişte kaç ürün var, dağılımı.

Kargo limit tarihinin analiz edilmesi: Teslim süreleri, geç teslimler.

Fiyat ve kargo değeri ilişkisi: Ürün fiyatı ile kargo fiyatı arasında korelasyon.

Zaman serisi analizi: Eğer tarih varsa, zaman içinde satış trendleri.

1. Ürün bazlı analizler
En çok satılan ürünler (product_id bazında satış adedi)

Ürünlerin ortalama fiyatları ve kargo ücretleri

En çok satılan ürünlerin hangi satıcıdan geldiği

2. Satıcı bazlı performans
Satıcıların toplam satış hacmi (ürün adedi ve tutar bazında)

Satıcıların ortalama ürün fiyatları ve kargo bedelleri

Satıcıların teslimat sürelerine bağlı analiz (shipping_limit_date ile ilişkilendirerek)

3. Sipariş bazlı analiz
Bir siparişte ortalama ürün sayısı (order_item_id kullanarak)

Siparişlerin toplam fiyat ve kargo bedeli (sipariş bazında price ve freight_value toplamı)

4. Zaman serisi ve teslimat analizleri
shipping_limit_date tarihlerini kullanarak siparişlerin gönderim süreleri veya gecikme analizi (bu tarihleri order date ile ilişkilendirerek)

Zaman içinde ürün satış trendleri

5. Fiyat & Kargo ilişkisi
Fiyat ve kargo bedeli arasındaki korelasyon

Fiyat aralıklarına göre kargo bedelinin değişimi

In [36]:
df_reviews.head()


review_id                          order_id  \
0  7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   
1  80e641a11e56f04c1ad469d5645fdfde  a548910a1c6147796b98fdf73dbeba33   
2  228ce5500dc1d8e020d8d1322874b6f0  f9e4b658b201a9f2ecdecbb34bed034b   
3  e64fb393e7b32834bb789ff8bb30750e  658677c97b385a9be170737859d3511b   
4  f7c4243c7fe1938f181bec41a392bdeb  8e6bfb81e283fa7e4f11123a3fb894f1   

   review_score review_comment_title  \
0             4                  NaN   
1             5                  NaN   
2             5                  NaN   
3             5                  NaN   
4             5                  NaN   

                              review_comment_message review_creation_date  \
0                                                NaN  2018-01-18 00:00:00   
1                                                NaN  2018-03-10 00:00:00   
2                                                NaN  2018-02-17 00:00:00   
3              Recebi bem antes do prazo estipulado.  2017-04-21 00:00:00   
4  Parabéns lojas lannister adorei comprar pela I...  2018-03-01 00:00:00   

  review_answer_timestamp  
0     2018-01-18 21:46:59  
1     2018-03-11 03:05:13  
2     2018-02-18 14:36:24  
3     2017-04-21 22:02:06  
4     2018-03-02 10:26:53

In [37]:
df_reviews.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99224 entries, 0 to 99223
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   review_id                99224 non-null  object
 1   order_id                 99224 non-null  object
 2   review_score             99224 non-null  int64 
 3   review_comment_title     11568 non-null  object
 4   review_comment_message   40977 non-null  object
 5   review_creation_date     99224 non-null  object
 6   review_answer_timestamp  99224 non-null  object
dtypes: int64(1), object(6)
memory usage: 5.3+ MB


In [38]:
check_missing_values(df_reviews)

Eksik Değer Sayısı  Yüzde (%)
review_comment_title                 87656      88.34
review_comment_message               58247      58.70

Eksik veriler:

review_comment_title ve review_comment_message alanlarında büyük oranda eksik yorum metinleri var. Bu, müşteri sadece puan vermiş ama yazılı yorum bırakmamış olabilir.

review_answer_timestamp çoğunlukla boş; satıcı her yoruma cevap vermeyebilir.

Dönüştürme yapılabilir:

review_creation_date ve review_answer_timestamp sütunları tarih formatına çevrilmeli (eğer string ise).

review_score kategorik değil ama sayı; puanları kategorik gibi gruplandırabiliriz (örneğin: 1-2 olumsuz, 3 nötr, 4-5 olumlu).

Eksik yorum metinleri NaN ya da boş string ise, metin analizinde bunlar ayrı kategoride işlenmeli.

Bireyler (unique):

review_id her satır için eşsiz (benzersiz).

order_id siparişler bazında bağlanabilir, bir siparişin birden fazla yorumu olabilir (kontrol edilmeli).

3. Bu Veri Setinden Yapılabilecek Analizler ve Sunumlar
Müşteri memnuniyeti dağılımı:

Puanların dağılımı, ortalama puan, medyan gibi temel istatistikler.

Zaman içinde müşteri puanlarının değişimi (örneğin aylık ortalama puan trendi).

Eksik yorum metinlerinin analizi:

Yorum yazan ve yazmayan müşterilerin oranı.

Yazılı yorumlarda sık geçen kelimeler, kelime bulutu.

Olumlu (4-5 puan) ve olumsuz (1-2 puan) yorumlardaki ortak kelimeler.

Satıcı cevap oranı:

Yorumlara satıcının ne sıklıkla cevap verdiği.

Yanıt verilen ve verilmemiş yorumlar arasında puan farkı.

Olumsuz yorumlar ve çözüm önerileri:

1-2 puan alan yorumlar üzerine detaylı metin incelemesi (hizmet, ürün, teslimat vb. problemler).

Zaman içinde olumsuz yorumların azalması artması.

Sipariş ve yorum ilişkisi:

Belirli ürün veya satıcı bazında puanların karşılaştırılması (başka veri setleriyle birleştirerek).

Sunumda:

Puan histogramları, puanların zamanla değişimi grafikleri, metin analizi sonuçları (kelime bulutu), satıcı yanıt oranı, eksik yorum oranı gibi grafiklerle desteklenebilir.

In [39]:
df_products.head()

product_id  product_category_name  \
0  1e9e8ef04dbcff4541ed26657ea517e5             perfumaria   
1  3aa071139cb16b67ca9e5dea641aaa2f                  artes   
2  96bd76ec8810374ed1b65e291975717f          esporte_lazer   
3  cef67bcfe19066a932b7673e239eb23d                  bebes   
4  9dc1a7de274444849c219cff195d0b71  utilidades_domesticas   

   product_name_lenght  product_description_lenght  product_photos_qty  \
0                 40.0                       287.0                 1.0   
1                 44.0                       276.0                 1.0   
2                 46.0                       250.0                 1.0   
3                 27.0                       261.0                 1.0   
4                 37.0                       402.0                 4.0   

   product_weight_g  product_length_cm  product_height_cm  product_width_cm  
0             225.0               16.0               10.0              14.0  
1            1000.0               30.0               18.0              20.0  
2             154.0               18.0                9.0              15.0  
3             371.0               26.0                4.0              26.0  
4             625.0               20.0               17.0              13.0

In [40]:
df_products.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32951 entries, 0 to 32950
Data columns (total 9 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   product_id                  32951 non-null  object 
 1   product_category_name       32341 non-null  object 
 2   product_name_lenght         32341 non-null  float64
 3   product_description_lenght  32341 non-null  float64
 4   product_photos_qty          32341 non-null  float64
 5   product_weight_g            32949 non-null  float64
 6   product_length_cm           32949 non-null  float64
 7   product_height_cm           32949 non-null  float64
 8   product_width_cm            32949 non-null  float64
dtypes: float64(7), object(2)
memory usage: 2.3+ MB


In [41]:
check_missing_values(df_products)

Eksik Değer Sayısı  Yüzde (%)
product_category_name                      610       1.85
product_name_lenght                        610       1.85
product_description_lenght                 610       1.85
product_photos_qty                         610       1.85
product_weight_g                             2       0.01
product_length_cm                            2       0.01
product_height_cm                            2       0.01
product_width_cm                             2       0.01

Entries: (örneğin) 32951 entries, 9 columns

Columns:

product_id (object) - eşsiz ürün kimliği, null yok

product_category_name (object) - ürün kategorisi, bazı eksikler olabilir

product_name_length (int64) - ürün isminin uzunluğu, null yok

product_description_length (int64) - ürün açıklamasının uzunluğu, null yok

product_photos_qty (int64) - ürün fotoğraf sayısı, null yok

product_weight_g (int64) - ürünün ağırlığı gram cinsinden, null olabilir

product_length_cm (int64) - ürün uzunluğu cm, null olabilir

product_height_cm (int64) - ürün yüksekliği cm, null olabilir

product_width_cm (int64) - ürün genişliği cm, null olabilir

Muhtemel eksik veriler:

product_category_name — bazen kategori boş olabilir

product_weight_g, product_length_cm, product_height_cm, product_width_cm — fiziksel ölçülerde eksikler olabilir

4. Bu Verisetinden Neler Yapılabilir?
Ürün Kategorileri Analizi:

En çok hangi kategoride ürün var?

Kategori bazında ürün sayıları, çeşitlilik analizi.

product_category_name eksikse, product_description_length veya product_name_length kullanılarak tahmini kategori analizi yapılabilir.

Ürün Özellikleri ve Boyutları:

Ürün ağırlık ve boyutlarının dağılımı, hangi aralıklar yoğun?

Kategoriye göre ürün ölçü ve ağırlık ortalamaları.

Boyut eksik olan ürünlerin oranı ve bunların nasıl ele alınacağı.

Fotoğraf Sayısı:

Ürün başına ortalama fotoğraf sayısı nedir? Az fotoğraf olan ürünler var mı?

Fotoğraf sayısı ile satış/verimlilik ilişkisi araştırılabilir (başka verilerle birleştirildiğinde).

Ürün Tanımları ve Metin Uzunlukları:

Ürün adı ve açıklama uzunluklarının analizi. Çok kısa ya da çok uzun açıklamalı ürünler var mı?

Bu metin uzunlukları satışa etki eder mi?

Eksik Verilerin Yönetimi:

Fiziksel özelliklerde eksik olanları nasıl doldurabiliriz? Ortalama mı yoksa başka yöntemler?

Kategorisi olmayan ürünler nasıl sınıflandırılır?

Gelişmiş Analiz:

Ürünlerin satış performansı ile ürün özellikleri korelasyonu (başka dataset ile birleştirerek).

Kategori ve ürün özelliklerine göre segmentasyon.

In [42]:
df_sellers.head()


seller_id  seller_zip_code_prefix  \
0  3442f8959a84dea7ee197c632cb2df15                   13023   
1  d1b65fc7debc3361ea86b5f14c68d2e2                   13844   
2  ce3ad9de960102d0677a81f5d0bb7b2d                   20031   
3  c0f3eea2e14555b6faeea3dd58c1b1c3                    4195   
4  51a04a8a6bdcb23deccc82b0b80742cf                   12914   

         seller_city seller_state  
0           campinas           SP  
1         mogi guacu           SP  
2     rio de janeiro           RJ  
3          sao paulo           SP  
4  braganca paulista           SP

In [43]:
df_sellers.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3095 entries, 0 to 3094
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   seller_id               3095 non-null   object
 1   seller_zip_code_prefix  3095 non-null   int64 
 2   seller_city             3095 non-null   object
 3   seller_state            3095 non-null   object
dtypes: int64(1), object(3)
memory usage: 96.8+ KB


In [44]:
check_missing_values(df_sellers)

Empty DataFrame
Columns: [Eksik Değer Sayısı, Yüzde (%)]
Index: []

seller_id	Her satıcıya atanmış benzersiz kimlik.
seller_zip_code_prefix	Satıcının posta kodu öneki (tam kod değil, bölgesel analizler için yeterli).
seller_city	Satıcının bulunduğu şehir (küçük harflerle verilmiş).
seller_state	Satıcının bulunduğu eyalet (örneğin SP = São Paulo).
Hiçbir sütunda eksik veri yok → 3095 satırın tamamı dolu.
Dönüştürme/Temizleme Gereksinimi:
seller_city ve seller_state verileri temiz; ancak:

Şehir isimleri küçük harflerle yazılmış → Görselleştirme için baş harf büyük yapılabilir.

Harita görselleştirmelerinde kodların açılımı gerekebilir (örneğin "SP" → "São Paulo").

seller_zip_code_prefix sadece 5 haneli önek → Harita verileriyle tam eşleşmeyebilir.

Bu Veri Setinden Yapılabilecek Analizler:
Satıcı Sayısı:

Kaç farklı satıcı var? Hangi şehirde/eyalette yoğunlaşmışlar?

Bölgesel Dağılım:

Satıcılar hangi şehirlerde ve eyaletlerde yoğunlukta?

Harita üzerinde satıcı dağılımı (heatmap, choropleth map).

Satıcıların Siparişlerle İlişkisi (birleştirerek):

En çok sipariş alan satıcılar kimler?

Ortalama satış fiyatı en yüksek olan satıcılar?

Satıcı Başına Ortalama Ürün Sayısı:

Her satıcının kaç farklı ürünü var? (df_items ile birleşince)

Kargo Mesafesi/Tahmini Süre Analizi (df_geolocation ile):

Satıcının konumu ile müşterinin konumu arasındaki mesafeler.

In [45]:
df_geolocation.head()

geolocation_zip_code_prefix  geolocation_lat  geolocation_lng  \
0                         1037       -23.545621       -46.639292   
1                         1046       -23.546081       -46.644820   
2                         1046       -23.546129       -46.642951   
3                         1041       -23.544392       -46.639499   
4                         1035       -23.541578       -46.641607   

  geolocation_city geolocation_state  
0        sao paulo                SP  
1        sao paulo                SP  
2        sao paulo                SP  
3        sao paulo                SP  
4        sao paulo                SP

In [46]:
df_geolocation.info

<bound method DataFrame.info of          geolocation_zip_code_prefix  geolocation_lat  geolocation_lng  \
0                               1037       -23.545621       -46.639292   
1                               1046       -23.546081       -46.644820   
2                               1046       -23.546129       -46.642951   
3                               1041       -23.544392       -46.639499   
4                               1035       -23.541578       -46.641607   
...                              ...              ...              ...   
1000158                        99950       -28.068639       -52.010705   
1000159                        99900       -27.877125       -52.224882   
1000160                        99950       -28.071855       -52.014716   
1000161                        99980       -28.388932       -51.846871   
1000162                        99950       -28.070104       -52.018658   

        geolocation_city geolocation_state  
0              sao paulo          

In [47]:
check_missing_values(df_geolocation)

Empty DataFrame
Columns: [Eksik Değer Sayısı, Yüzde (%)]
Index: []

📄 Kolonlar:
geolocation_zip_code_prefix: Konuma ait posta kodu (yakınlaştırılmış / kısaltılmış format).

geolocation_lat: Enlem bilgisi.

geolocation_lng: Boylam bilgisi.

geolocation_city: Şehir adı.

geolocation_state: Eyalet / bölge kodu (örneğin: SP, RJ).

🧼 Eksik Değer Durumu:
Hiçbir kolonda eksik veri yok.

Bu, veri temizliği açısından olumlu bir durum.

🔄 Dönüştürülmesi / Dikkat Edilmesi Gerekenler:
Posta kodu seller_zip_code_prefix ve customer_zip_code_prefix ile eşleşerek ilişkilendirme yapılabilir.

Lat/Lng verileri harita üzerinde görselleştirme veya mesafe hesaplama gibi coğrafi analizler için kullanılabilir.

geolocation_city'de küçük harfli yazım var. Harita görselleştirmelerinde uyumlu hale getirmek için düzenlenebilir.

Aynı posta koduna ait birden fazla lat/lng kaydı olabilir → ortalama koordinatlar alınarak sadeleştirilebilir.

🔍 Bu Veriden Yapılabilecek Analizler:
📍 Satıcı / Müşteri konumlarını harita üzerinde görselleştirme
→ En yoğun bölgeler, dağılım haritaları, eyalet bazlı satış analizi.

🛣️ Satıcı ve müşteri arasındaki tahmini mesafe hesaplaması
→ Ortalama teslimat süresi vs. mesafe analizi.

🗺️ En çok müşteri/satıcı yoğunluğu olan şehirler ve eyaletler
→ Pazar stratejileri için bölgesel analiz yapılabilir.

🧭 Teslimat sürelerini coğrafi bölgelere göre karşılaştırma
→ Örneğin SP eyaletindeki siparişler RJ’ye göre daha hızlı mı ulaşıyor?

🔁 Kümelenme (Clustering) ile benzer bölgelerin tespiti
→ Teslimat performansı açısından bölgesel segmentasyon yapılabilir.

In [48]:
df_category_translation.head()

product_category_name product_category_name_english
0            beleza_saude                 health_beauty
1  informatica_acessorios         computers_accessories
2              automotivo                          auto
3         cama_mesa_banho                bed_bath_table
4        moveis_decoracao               furniture_decor

In [49]:
df_category_translation.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 71 entries, 0 to 70
Data columns (total 2 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   product_category_name          71 non-null     object
 1   product_category_name_english  71 non-null     object
dtypes: object(2)
memory usage: 1.2+ KB


In [50]:
check_missing_values(df_category_translation)

Empty DataFrame
Columns: [Eksik Değer Sayısı, Yüzde (%)]
Index: []

Kolonlar:
product_category_name
→ Portekizce ürün kategori adları.

product_category_name_english
→ Yukarıdaki kategorilerin İngilizce karşılıkları.

🧼 Eksik Değer Durumu:
Eksik veri bulunmamaktadır. Tüm satırlar dolu ve temiz.

🔄 Dönüştürülmesi / Dikkat Edilmesi Gerekenler:
Kategori adlarının diğer veri setleriyle eşleştirilmesi için bu tablo bir sözlük (mapping) gibi kullanılabilir.

Eğer df_products içinde Portekizce kategori adları varsa, bunlar İngilizce'ye çevrilebilir.

Bazı veri setlerinde product_category_name boş gelebilir → bu durumda çeviri yapılamaz, dikkat edilmesi gerekir.

🔍 Bu Veriden Yapılabilecekler:
🌍 Ürün analizlerini İngilizce kategori isimleriyle yapmak
→ Daha açıklayıcı grafikler ve sunumlar için çeviriyi uygulamak.

📊 Kategori bazlı satış / yorum / memnuniyet analizlerinde İngilizce adları kullanmak
→ Örneğin: "Which category has the most reviews?" analizinde.

🧾 Kategorileri birleştirme veya gruplayarak üst kategoriler oluşturmak
→ Örneğin: "health_beauty" ve "perfumery" gibi kategorileri tek çatı altında gruplamak.

📚 Çok kategorili veri setlerinde kullanıcı dostu dil kullanmak
→ Dashboard veya raporlar için daha anlaşılır kategori isimleri sunmak.

Bu veri seti yardımcı (referans) niteliğinde olduğundan, esas analizlerde anlamlı sonuçlar üretmek için df_products gibi veri setleriyle ilişkilendirilerek kullanılacaktır.



In [51]:
df_payments.head()

order_id  payment_sequential payment_type  \
0  b81ef226f3fe1789b1e8b2acac839d17                   1  credit_card   
1  a9810da82917af2d9aefd1278f1dcfa0                   1  credit_card   
2  25e8ea4e93396b6fa0d3dd708e76c1bd                   1  credit_card   
3  ba78997921bbcdc1373bb41e913ab953                   1  credit_card   
4  42fdf880ba16b47b59251dd489d4441a                   1  credit_card   

   payment_installments  payment_value  
0                     8          99.33  
1                     1          24.39  
2                     1          65.71  
3                     8         107.78  
4                     2         128.45

In [52]:
df_payments.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103886 entries, 0 to 103885
Data columns (total 5 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   order_id              103886 non-null  object 
 1   payment_sequential    103886 non-null  int64  
 2   payment_type          103886 non-null  object 
 3   payment_installments  103886 non-null  int64  
 4   payment_value         103886 non-null  float64
dtypes: float64(1), int64(2), object(2)
memory usage: 4.0+ MB


In [53]:
check_missing_values(df_payments)

Empty DataFrame
Columns: [Eksik Değer Sayısı, Yüzde (%)]
Index: []

📌 Kolonlar:
Sütun Adı	Açıklama
order_id	Siparişin benzersiz ID'si
payment_sequential	Aynı siparişe ait ödeme sıralaması (1, 2, ...)
payment_type	Ödeme yöntemi (credit_card, boleto, debit_card, voucher vs.)
payment_installments	Taksit sayısı
payment_value	Toplam ödeme miktarı (R$ - Brezilya Reali)

🔎 Genel Özellikler:
Toplam Satır Sayısı: 103,886

Eksik Veri: Yok ✅

Dönüştürülmesi Gereken Alan: payment_type → kategorik veri tipi yapılabilir.

order_id bazında aynı sipariş için birden fazla ödeme olabilir. Analizlerde sipariş bazlı toplam ödeme dikkatle ele alınmalı.

📊 Bu Veri Seti ile Yapılabilecek Analizler:
1. Ödeme Yöntemlerinin Dağılımı
Hangi ödeme yöntemleri en çok tercih ediliyor?

Pie chart veya bar chart ile görselleştirilebilir.

2. Ödeme Türüne Göre Ortalama Ödeme Tutarı
payment_type bazında mean(payment_value)

Kim daha çok harcıyor? Kredi kartı mı, banka kartı mı?

3. Taksit Sayısının Dağılımı
Ortalama kaç taksitle alışveriş yapılıyor?

En çok kaç taksit kullanılıyor?

4. Sipariş Başına Ödeme Analizi
Aynı order_id için birden fazla kayıt varsa groupby(order_id).sum(payment_value) yapılmalı.

5. Yüksek Tutar / Az Taksit Uç Değer Analizi
payment_value çok yüksek olup payment_installments çok düşük olanlar incelenebilir.

6. En Yaygın Kombinasyonlar
payment_type + payment_installments kombinasyonlarının frekansı.

7. Satış Gelirlerinin Kaynağı
Toplam ödeme miktarına göre en çok gelir getiren ödeme tipi.

Tüm bu analizler için istenirse orders veri seti ile birleştirme (join) yapılarak zaman bazlı, müşteri bazlı ya da coğrafi kırılımlı analizler de gerçekleştirilebilir.